### DISCUSSION TITLE
Feed published successfully but shows error on client

### DISCUSSION THREAD

**Discussion by chukaibejih** [MarshalX/atproto#62] [General] [answered]
Hello @MarshalX , Firstly I have to say you did a great job with this project. I am working on a custom feed using it.

Do you have any idea why I get an error on the app even though the feed was published and these endpoints work on my domain? 

```
- /.well-known/did.json
- /xrpc/app.bsky.feed.describeFeedGenerator
- /xrpc/app.bsky.feed.getFeedSkeleton

```
Although a similar issue was raised here on the official bluesky repo - https://github.com/bluesky-social/feed-generator/issues/35 but I could really use a second eye. 
![image](https://github.com/MarshalX/bluesky-feed-generator/assets/29266211/9658095a-bbce-42d5-a345-232069e36af6)

**Comment by MarshalX** [comment]
Hi. Thank you ;)

Could you send your domain name and provide link to the published feed?

**Comment by chukaibejih** [comment]
Thank you for the swift response. 
Here are the links you requested 
- https://bsky.app/profile/did:plc:76il4ns5xbqbsqjwibjgcnsk/feed/epl-buzz
- https://epl-feed.onrender.com/
- https://epl-feed.onrender.com/.well-known/did.json
- https://epl-feed.onrender.com/xrpc/app.bsky.feed.describeFeedGenerator
- https://epl-feed.onrender.com/xrpc/app.bsky.feed.getFeedSkeleton?feed=at://did:plc:76il4ns5xbqbsqjwibjgcnsk/app.bsky.feed.generator/epl-buzz

I provided all the possible links related to the feed. Please let me know if you need any other info.

**Comment by MarshalX** [comment]
> Thank you for the swift response. Here are the links you requested
> 
>     * https://bsky.app/profile/did:plc:76il4ns5xbqbsqjwibjgcnsk/feed/epl-buzz
> 
>     * https://epl-feed.onrender.com/
> 
>     * https://epl-feed.onrender.com/.well-known/did.json
> 
>     * https://epl-feed.onrender.com/xrpc/app.bsky.feed.describeFeedGenerator
> 
>     * https://epl-feed.onrender.com/xrpc/app.bsky.feed.getFeedSkeleton?feed=at://did:plc:76il4ns5xbqbsqjwibjgcnsk/app.bsky.feed.generator/epl-buzz
> 
> 
> I provided all the possible links related to the feed. Please let me know if you need any other info.

First of all your server boots too slow. Maybe you are using some free service that turns off your webserver due to inactivity. This could be a problem. 

The second thing is could you tell me how did you publish your feed?

**Comment by MarshalX** [comment]
@chukaibejih it doesn't looks legit. fix it first 

<img width="1280" alt="image" src="https://github.com/MarshalX/bluesky-feed-generator/assets/15520314/a052d355-f584-4fec-9f32-711834456c98">

**Comment by chukaibejih** [comment]
- About the server, you are right. I am using a free tier because I wanted to get it working first before going for a paid plan but you could recommend a good one that's not so pricey 

- About publishing the feed, I basically followed the readme and made the changes in the algos and publisherfeed files. Set up the necessary env variables then ran the publisherfeed command.  

- Once the feed was published, I deployed the project and tested those three endpoints. 

- I checked the Bluesky app and saw the feed name on my profile which was the screenshot I shared with you. 

But from your assessment on those links, are they returning the appropriate response and expected?

**Comment by MarshalX** [comment]
> * About the server, you are right. I am using a free tier because I wanted to get it working first before going for a paid plan but you could recommend a good one that's not so pricey
> 
>     * About publishing the feed, I basically followed the readme and made the changes in the algos and publisherfeed files. Set up the necessary env variables then ran the publisherfeed command.
> 
>     * Once the feed was published, I deployed the project and tested those three endpoints.
> 
>     * I checked the Bluesky app and saw the feed name on my profile which was the screenshot I shared with you.
> 
> 
> But from your assessment on those links, are they returning the appropriate response and expected?

the endpoints are fine. but published feed is wrong imho

**Comment by MarshalX** [comment, marked as answer]
> @chukaibejih it doesn't looks legit. fix it first
> <img alt="image" width="1280" src="https://user-images.githubusercontent.com/15520314/243728984-a052d355-f584-4fec-9f32-711834456c98.png">

@chukaibejih look at the correct feed:
<img width="1297" alt="image" src="https://github.com/MarshalX/bluesky-feed-generator/assets/15520314/d652e068-65b2-4d3f-8e42-5939ab510d8a">

**Comment by chukaibejih** [comment]
> @chukaibejih it doesn't looks legit. fix it first 
> 
> <img width="1280" alt="image" src="https://github.com/MarshalX/bluesky-feed-generator/assets/15520314/a052d355-f584-4fec-9f32-711834456c98">
> 

I assume you are referring to this yea  `did=' did:web: Big-0le',`

Big-Ole is my PC name. I wonder how it got there. I will look into it. Probably republish the feed

**Reply by chukaibejih** [reply]
@MarshalX I was able to fix the did issue which you pointed out and I now have a new error on the client 
![image](https://github.com/MarshalX/atproto/assets/29266211/9ac5c5d3-7b2f-45b0-83d6-672a513e0969)

![image](https://github.com/MarshalX/atproto/assets/29266211/b0208663-9b59-4fa8-a520-5dc5c7d97c1e)

**Reply by MarshalX** [reply]
> @MarshalX I was able to fix the did issue which you pointed out and I now have a new error on the client 
> 
> ![image](https://github.com/MarshalX/atproto/assets/29266211/9ac5c5d3-7b2f-45b0-83d6-672a513e0969)
> 
> 
> 
> ![image](https://github.com/MarshalX/atproto/assets/29266211/b0208663-9b59-4fa8-a520-5dc5c7d97c1e)
> 
> 
> 
> 

Because your did.json endpoint returns wrong content. Few hours ago is was fine

Idk where did you get it 

> "id":"did:web:srv-chu4l2vdvk4olivc0730-hibernate-5bd99f785c-r4z5r"

**Reply by chukaibejih** [reply]
I am obviously messing up something somewhere. Let me go through my codebase again in detail. I really appreciate your help!

**Reply by chukaibejih** [reply]
IT WORKED HAHAHA! All I did was update the HOSTNAME on Render to what I used to publish the feed. I am so glad, Thank you so much. The next thing to do is to work on improving the feed algorithm because I see posts that are not even football-related.
![image](https://github.com/MarshalX/atproto/assets/29266211/f77ddf27-7fc8-4509-bdc1-28de33a398a5)
